### PR TITLE
DP-3711 Add example for InferDateTimeFromCSVFile

### PR DESCRIPTION
**Pull request by akmalxpert**

### CODE REVIEW

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/InferDateTimeFromCSVFile.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.northconcepts.datapipeline.foundations.time.DateTimePattern;
+
+import java.util.Map;
+import java.util.concurrent.atomic.LongAdder;
+
+public class InferDateTimeFromCSVFile {
+    public static void main(String[] args) {
+        CsvPipelineInput pipelineInput = new CsvPipelineInput()
+                .setFileSource(new LocalFileSource().setPath("example/data/input/DateTimeData.csv"))
+                .setFieldNamesInFirstRow(true);
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/InferDateTimeFromCSVFile.java]: Let's not use CsvPipelineInput.  Instead, let's use something simpler like `pipeline.setInputAsDataReaderFactory(() -> new CSVReader("example/data/input/DateTimeData.csv").setFieldNamesInFirstRow(true));`.

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/InferDateTimeFromCSVFile.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        Pipeline pipeline = new Pipeline();
+        pipeline.setInput(pipelineInput);
+
+        Dataset dataset = new MemoryDataset(pipeline);
+        dataset.load().waitForColumnStatsToLoad();
+        dataset.setCollectUniqueValues(true);
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/InferDateTimeFromCSVFile.java]: `setCollectUniqueValues(true)` should be called before `load()`, but do we really need it set to true?

**Review by DeleTaylor — CHANGES_REQUESTED** [review]

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/InferDateTimeFromCSVFile.java (hunk 1 above)
Comment (akmalxpert) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/InferDateTimeFromCSVFile.java]: done

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/InferDateTimeFromCSVFile.java (hunk 2 above)
Comment (akmalxpert) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/InferDateTimeFromCSVFile.java]: I removed this line

**Review by DeleTaylor — APPROVED** [review]

**Pull request merged**